contracts/core/price-aggregator/src/events.rs

```diff
@@ -40,7 +49,53 @@ pub trait EventsModule {
         &self,
         #[indexed] from: &ManagedBuffer,
         #[indexed] to: &ManagedBuffer,
-        #[indexed] epoch: u64,
+        #[indexed] round: usize,
         new_round_event: &NewRoundEvent<Self::Api>,
     );
+
+    fn emit_discard_submission_event(
+        &self,
+        token_pair: &TokenPair<Self::Api>,
+        round_id: usize,
+        submission_timestamp: u64,
+        first_submission_timestamp: u64,
+        has_caller_already_submitted: bool,
+    ) {
```

Comment:
Don't have functions with more than 5 arguments, including `&self`, otherwise, you can run into weird issues. Group them in a struct and pass that.

Reply:
Those issues are gone, they were stemming from wasmer. Readability-wise structures are still nice